model/annotations/task_annotations.go

```diff
@@ -146,3 +146,40 @@ func RemoveSuspectedIssueFromAnnotation(taskId string, execution int, issue Issu
 		bson.M{"$pull": bson.M{SuspectedIssuesKey: issue}},
 	)
 }
+
+func UpdateAnnotation(a *TaskAnnotation, userDisplayName string) error {
+	source := &Source{
+		Author:    userDisplayName,
+		Time:      time.Now(),
+		Requester: APIRequester,
+	}
+	update := bson.M{}
+
+	if a.Metadata != nil {
+		update[MetadataKey] = a.Metadata
+	}
+	if a.Note != nil {
+		a.Note.Source = source
+		update[NoteKey] = a.Note
+	}
+	if a.Issues != nil {
+		for _, i := range a.Issues {
+			i.Source = source
+		}
+		update[IssuesKey] = a.Issues
+	}
+	if a.SuspectedIssues != nil {
+		for _, i := range a.SuspectedIssues {
+			i.Source = source
+		}
+		update[SuspectedIssuesKey] = a.SuspectedIssues
+	}
+	_, err := db.Upsert(
```

Comment:
Trying to think of edge cases; what if the annotation doc given doesn't have anything to update? (i.e. len(update) == 0) I think then we shouldn't upsert, we should maybe return an error "nothing to update" or something (or return nil? I'm not super picky about what the behavior is, just want to make sure it's been thought about).

Reply:
good point! thanks!